tmt/steps/prepare/install.py

```diff
@@ -26,44 +43,110 @@ def options(cls, how=None):
         """ Prepare command line options """
         return [
             click.option(
-                '-p', '--package', metavar='NAME', multiple=True,
-                help='Package name to be installed.'),
+                '-p', '--package', metavar='PACKAGE', multiple=True,
+                help='Package name or path to rpm to be installed.'),
+            click.option(
+                '-D', '--directory', metavar='PATH', multiple=True,
+                help='Path to a local directory with rpm packages.'),
             click.option(
                 '-c', '--copr', metavar='REPO', multiple=True,
                 help='Copr repository to be enabled.')
             ] + super().options(how)
 
     def show(self):
         """ Show provided scripts """
-        super().show(['package', 'copr'])
+        super().show(['package', 'directory', 'copr'])
 
     def wake(self, data=None):
         """ Override options and wake up the guest """
-        super().wake(['package', 'copr'])
+        super().wake(['package', 'directory', 'copr'])
 
         # Convert to list if necessary
-        tmt.utils.listify(self.data, split=True, keys=['package', 'copr'])
+        tmt.utils.listify(
+            self.data, split=True, keys=['package', 'directory', 'copr'])
 
     def go(self, guest):
         """ Prepare the guests """
         super().go()
+        # Nothing to do in dry mode
+        if self.opt('dry'):
+            return
+
+        # Prepare the right dnf/yum command
+        self.debug('Check if sudo is necessary.', level=2)
+        user = guest.execute('whoami')[0].strip()
+        sudo = '' if user == 'root' else 'sudo '
+        self.debug('Check if dnf is available.', level=2)
+        try:
+            guest.execute('rpm -q dnf')
+            command = f"{sudo}dnf"
+            plugin = 'dnf-plugins-core'
+        except tmt.utils.RunError:
+            command = f"{sudo}yum"
+            plugin = 'yum-plugin-copr'
```

Comment:
This won't work on rhel6/centos6, afaik there is no such package. Could we add a check somewhere in provision to abort on centos6 and smilar

Reply:
But you can still install packages from non-copr repositories, which works fine:

```
> tmt run -a plan --name smoke provision -h container -i centos:6 prepare -h install -p gcc
/var/tmp/tmt/run-202

/plans/smoke
    discover
        how: shell
        directory: /home/psss/git/tmt
        summary: 1 test selected
    provision
        how: container
        image: centos:6
        distro: CentOS release 6.10 (Final)
        kernel: 5.5.11-200.fc31.x86_64
        summary: 1 guest provisioned
    prepare
        how: install
        package: gcc
        summary: 1 preparation applied
    execute
        how: shell
        summary: 1 test executed
    report
        how: display
        summary: 1 error
    finish
        container: stopped
        container: removed
        summary: 0 tasks completed
```

Why should we completely abort on rhel6/centos6?

Reply:
@psss sorry we need also support centos 6 & friends :( `convert2rhel` already uses this :(

https://github.com/oamg/convert2rhel/pull/40

 for centos 6 we can install copr builds by manually adding the repo like this:

https://gitlab.com/testing-farm/cruncher/-/blob/master/modules/cruncher.py#L399

Reply:
Hm, there will need to be some special handling for group coprs:

* https://copr.fedorainfracloud.org/coprs/g/oamg/convert2rhel/repo/epel-6/group_oamg-convert2rhel-epel-6.repo

The url is different, contains an extra `/g/` and `group_` keys.

Reply:
Ok, added suppport and test for `epel6` as well.